foreshadow/transformers/transformers.py

```diff
@@ -58,187 +92,288 @@ def wrap_transformer(transformer):
+
+            ..#noqa: I102
+
+            """
+            self.keep_columns = keep_columns
+            self.name = name
+            super(DFTransformer, self).__init__(*args, **kwargs)
+
+        def get_params(self, deep=True):
+            """Override standard get_params to handle nonstandard init.
+
+            BaseEstimator for sklearn gets and sets parameters based on the
+            init statement for that class. Since this class is used to wrap
+            a parent transformer (by OOP), we use the parent's init
+            statement and then this DFTransformer's additional arguments.
+            We must override of BaseEstimator will complain about our
+            nonstandard usage.
+
+            Args:
+                deep (bool): If True, will return the parameters for this
+                    estimator and contained subobjects that are estimators.
+
+            Returns:
+                Parameter names mapped to their values for parent +
+                DFTransformer wrapper.
+
+            """
+            parent_params = BaseEstimator.get_params(transformer, deep=deep)
+            # will contain any init arguments that are not variable keyword
+            # arguments. By default, this means that any new transformer
+            # cannot have variable keyword arguments in its init less the
+            # transformer designer is okay with it not getting picked up here.
+            # The transformer class passed will not contain the current values,
+            # so we set them with the values on the object instance, below.
+            self_params = dict()  # the output
+            init = getattr(self.__init__, "deprecated_original", self.__init__)
+            if init is object.__init__:
+                return self_params
+            # explicit constructor to introspect
+            # introspect the constructor arguments to find the model
+            # parameters to represent
+            init_signature = signature(init)
+            # Consider the constructor parameters excluding 'self'
+            self_sig = [
+                p
+                for p in init_signature.parameters.values()
+                if p.name != "self"
+                and p.kind != p.VAR_KEYWORD
+                and p.kind != p.VAR_POSITIONAL
+            ]
+            self_sig = sorted([p.name for p in self_sig])
+            for key in self_sig + list(parent_params.keys()):
+                warnings.simplefilter("always", DeprecationWarning)
+                try:
+                    with warnings.catch_warnings(record=True) as w:
+                        value = getattr(self, key, None)
+                    if len(w) and w[0].category == DeprecationWarning:
+                        # if the parameter is deprecated, don't show it
+                        continue
+                finally:
+                    warnings.filters.pop(0)
+
+                # XXX: should we rather test if instance of estimator?
```

Comment:
was this addressed? or should we remove this comment?

Reply:
sklearn comment to put here for us to keep track of

Reply:
comment from sklearn